### ISSUE TITLE
TensorFlowV2Classifier and KerasClassifier do not validate labels when computing loss

### ISSUE THREAD

**Issue by f4str** [Trusted-AI/adversarial-robustness-toolbox#1959]
**Describe the bug**
In the current implementations for the `TensorFlowV2Classifier` and `KerasClassifier`, when calling the `compute_loss(x, y)` method, the labels are not validated and transformed. This causes an error when the `argmax` function is called and integer class labels are passed.

The `PyTorchClassifier` class performs this validation so the error is avoided. The fix is relatively straightforward, just call the `check_and_transform_label_format` function in the `TensorFlowV2Classifier` and `KerasClassifier` classes just like is done with the PyTorch variation.

**To Reproduce**
Instantiate a `TensorFlowV2Classifer` or `KerasClassifier` instance using the `SparseCategoricalCrossentropy` loss. Call the `compute_loss(x, y)` function and use labels for `y` (e.g., shape `(n,)`) and there will be an error due the argmax function call.

**Expected behavior**
The labels should be validated and transformed accordingly.

**System information (please complete the following information):**
 - OS
 - Python version
 - ART version or commit number
 - TensorFlow / Keras / PyTorch / MXNet version
